xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
That sounds good, though I think ill stick with just X11/DRM atm and once the Mir branch lands I can purpose a change for Mir (same with wayland since I dont think its around)